packages/android_intent/android/src/main/java/dev/flutter/plugins/androidintent/AndroidIntentPlugin.java

```diff
@@ -0,0 +1,66 @@
+package dev.flutter.plugins.androidintent;
+
+import androidx.annotation.NonNull;
+import io.flutter.embedding.engine.plugins.FlutterPlugin;
+import io.flutter.embedding.engine.plugins.activity.ActivityAware;
+import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding;
+import io.flutter.plugin.common.MethodChannel;
+import io.flutter.plugin.common.PluginRegistry.Registrar;
+
+/**
+ * The new embedding implementation of the plugin.
+ *
+ * <p>This can be included in an add to app scenario to gracefully handle activity and context
+ * changes, unlike the previous {@link io.flutter.plugins.androidintent.AndroidIntentPlugin}.
+ */
+public class AndroidIntentPlugin implements FlutterPlugin, ActivityAware {
+  private final IntentSender sender;
+
+  /**
+   * Initialize this within the {@code #configureFlutterEngine} of a Flutter activity or fragment.
+   *
+   * <p>See {@code dev.flutter.plugins.androidintentexample.MainActivity} for an example.
+   */
+  public AndroidIntentPlugin() {
+    sender = new IntentSender(/*activity=*/ null, /*context=*/ null);
+  }
+
+  /** This exists for legacy compatibility purposes. */
+  public static void registerWith(Registrar registrar) {
+    io.flutter.plugins.androidintent.AndroidIntentPlugin.registerWith(registrar);
+  }
+
+  @Override
+  public void onAttachedToEngine(@NonNull FlutterPluginBinding binding) {
+    sender.setApplicationContext(binding.getApplicationContext());
```

Comment:
To clarify, do we use the application context when it's available because devs might be sending Intents with data that don't launch an `Activity`? Because launching an `Activity` from an application context requires special window settings.

Reply:
We do set `FLAG_ACTIVITY_NEW_TASK` when launching from just the application context, is there something else we should be doing? I'm following the existing logic in the plugin here, which may be buggy.

I'm not 100% sure of the original intent. I'm guessing that it's more supposed to be a safeguard in cases where there is no activity available for whatever reason than a deliberate option for developers.

Reply:
Ok. Probably needs to be there then. But generally we'd like devs to understand the consequences of launching a new task.